### ISSUE TITLE
`invalid-method-override` false positive when the base method annotates `self` with a Protocol (regression 0.0.59 → 0.0.60)

### ISSUE THREAD

**Issue by SebbyLaw** [astral-sh/ty#4018]
## Summary

Since 0.0.60, overriding a method whose base definition annotates `self` with a Protocol type is flagged `invalid-method-override` — even when the override's signature is **byte-for-byte identical** to the base, and even when the subclass structurally satisfies the protocol. 0.0.59 accepts all of these.

The pattern is a plain mixin class that types `self` against a protocol which the final concrete class implements (so the mixin body can use protocol members). An identical-signature override cannot violate LSP, so at least that case is a false positive.

## Reproduction

https://play.ty.dev/cee15557-156e-4d21-874e-1c405f5cf4cc

```python
from typing import Protocol


class P(Protocol):
    def __contains__(self, key: str) -> bool: ...
    def m(self) -> None: ...


class MixinA:
    # mixin method typed against the protocol, class does NOT satisfy P
    def m(self: P) -> None: ...


class Sub(MixinA):
    # identical signature to the base
    def m(self: P) -> None: ...  # error[invalid-method-override]: Invalid override of method `m`


class SubPlain(MixinA):
    # implicit self
    def m(self) -> None: ...  # error[invalid-method-override]: Invalid override of method `m`


class SatisfiesP(MixinA):
    def __contains__(self, key: str) -> bool:
        return False

    def m(self: P) -> None: ...  # error[invalid-method-override]: Invalid override of method `m`
```

### Version

ty 0.0.60 (ae554ec78 2026-07-16)
